Backend/server/src/main/java/com/uaic/server/controller/BusinessController.java

```diff
@@ -212,4 +218,34 @@ public ResponseEntity<Void> deleteEmployeeFromBusiness(@PathVariable UUID busine
         }
     }
 
+    @GetMapping("/{businessId}/posts")
+    public ResponseEntity<Iterable<PostOutDTO>> getPostsByBusiness(@PathVariable UUID businessId) {
+
+        UserOutDTO userInfo = userService.getAuthenticatedUserInfo();
+
+        Optional<Business> optionalBusiness = businessService.findBusinessById(businessId);
+        if (!optionalBusiness.isPresent()) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+
+        Business business = optionalBusiness.get();
+        Optional<Employee> optionalConnectedEmployee = employeeService.findEmployeeByEmail(userInfo.getEmail());
+        if (!optionalConnectedEmployee.isPresent() ||
+                (optionalConnectedEmployee.isPresent()
+                        && !optionalConnectedEmployee.get().getBusiness().equals(business))) {
```

Comment:
sa adaugi un comentariu simplu gen "// to be changed when solving businessId param redundancy" ca sa tinem minte sa modificam cand rezolvam faza cu businessId ca parametru fiind useless